src/Settings/ConfigurationPreparationService.php

```diff
@@ -37,9 +38,14 @@ public function prepareConfiguration(string $configName, $currentConfig = null,
             }
             $config = $currentConfig;
         } else {
-            $configuration = Dao::getByName($configName);
+            $configuration = Configuration::getByName($configName);
             if (!$configuration) {
-                throw new \Exception('Configuration ' . $configName . ' does not exist.');
+                throw new InvalidArgumentException(
```

Comment:
Shouldn't we use `Symfony\Component\HttpKernel\Exception\NotFoundHttpException` here?